x/market/keeper/deal.go

```diff
@@ -138,14 +138,36 @@ func (k Keeper) SellOwnData(ctx sdk.Context, seller sdk.AccAddress, cert types.D
 	}
 
 	if findDeal.GetStatus() != ACTIVE {
-		return sdk.Coin{}, fmt.Errorf("deal is not activated")
+		return sdk.Coin{}, sdkerrors.Wrap(err, "deal is not activated")
 	}
 
 	dealAddress, err := types.AccDealAddressFromBech32(findDeal.GetDealAddress())
 	if err != nil {
 		return sdk.Coin{}, err
 	}
 
+	// Is trusted validator
+	validator := cert.UnsignedCert.GetDataValidatorAddress()
+	trustedValidators := findDeal.GetTrustedDataValidators()
+
+	flag := false
+	for _, v := range trustedValidators {
+		if validator == v {
+			flag = true
+			break
+		}
+	}
+	if !flag {
+		return sdk.Coin{}, sdkerrors.Wrap(err, "data validator is invalid address")
+	}
+
+	// Is deal's balance insufficient
+	dealBalance := k.bankKeeper.GetBalance(ctx, dealAddress, "umed")
+	sellerBalance := k.bankKeeper.GetBalance(ctx, seller, "umed")
+	if dealBalance.IsLT(sellerBalance) {
+		return sdk.Coin{}, fmt.Errorf("deal's balance is smaller than seller's balance")
+	}
```

Comment:
As I understand, this code block should be deleted. Below, you're already checking whether `isTrustedValidate(...)` and `dealBalance < pricePerData`.

Reply:
Thanks for your feedback. I think these code blocks are created when fix base branch. 😭 I'll change it!

Reply:
No worries. I guessed so. :)